tests/examples/common.py

```diff
@@ -0,0 +1,63 @@
+import os
+import tempfile
+import subprocess
+
+MOCOCRW_EXAMPLE_BINARY_PATH = "."
+try:
+    MOCOCRW_EXAMPLE_BINARY_PATH = os.environ["MOCOCRW_EXAMPLE_BINARY_PATH"]
+except KeyError:
+    print("Please set the MOCOCRW_EXAMPLE_BINARY_PATH environment variable.")
+    exit(1)
+
+MOCOCRW_DIR = ""
+try:
+    MOCOCRW_SRC_DIR = os.environ["MOCOCRW_SRC_DIR"]
+except KeyError:
+    print("Please set the MOCOCRW_SRC_DIR environment variable")
+    exit(1)
+
+
+HASH_ALGOS = [
+    "SHA256", "SHA384", "SHA512", "SHA3-256", "SHA3-384", "SHA3-512"
+]
+
+
+def add_invalid_parameters_and_execute(execute_string):
+    def execute(exec_string):
+        result = subprocess.run(exec_string.split(), capture_output=True)
+        assert result.returncode == 1
+        assert b"Unknown option selected: unrecognised option '--aw\xc3\xb6eljf'\n" \
+               b"Failure parsing command line.\n" == result.stderr
+    execute(execute_string + " --awöeljf asdölvkjyxc")
```

Comment:
I would prefer if we didn't use strings but lists of arguments here so we can avoid another shell indirection (which may be required for this to work in a cross compilation environment anyway):

```suggestion
def add_invalid_parameters_and_execute(parameters):
    parameters.extend(["--awöeljf", "asdölvkjyxc"])
    result = subprocess.run(parameters, capture_output=True)
    assert result.returncode == 1
    assert ...
```